exercises/flatten-array/FlattenArrayExample.swift

```diff
@@ -0,0 +1,37 @@
+import Foundation
+
+func flattenArray(list: [Any?]) -> [Any] {
```

Comment:
Specifically It would require Conditional Conformances which are not yet implemented 
https://github.com/apple/swift-evolution/blob/master/proposals/0143-conditional-conformances.md